src/GherkinCompatibilityMode.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Gherkin;
+
+enum GherkinCompatibilityMode: string
+{
+    case LEGACY = 'legacy';
+
+    case GHERKIN_32 = 'gherkin-32';
+
+    public function shouldRemoveFeatureDescriptionPadding(): bool
```

Comment:
should we mark all those methods as internal ?

To me, the only public API of the enum should be the list of cases.

Reply:
Good suggestion.

I'd thought of leaving it public as it may be useful e.g. for Behat to know whether padding was removed or not (to then know whether to add it back).

But actually I think we should try to solve any of these downstream without reference to the parsing mode (e.g. we can just look at whether there is / is not whitespace at the start of the first line).

Then it will be easier to remove these if & when they're no longer required.